pkgs/os-specific/linux/kernel/grsecurity-config.nix

```diff
@@ -0,0 +1,117 @@
+{ stdenv, version }:
+
+with stdenv.lib;
+
+''
+GCC_PLUGINS y
+
+# Auto configuration with these constraints will enable most of the
+# important features (RAP, UDEREF, ASLR, memory sanitization).
+GRKERNSEC y
+GRKERNSEC_CONFIG_CUSTOM y # do custom configuration
+
+PAX y
+
+# PaX control
+PAX_SOFTMODE y
+PAX_PT_PAX_FLAGS y
+PAX_XATTR_PAX_FLAGS y
+PAX_EI_PAX n
+
+# PaX Nonexecutable Pages
+PAX_NOEXEC y
+PAX_PAGEEXEC y
+PAX_EMUTRAMP y
+PAX_MPROTECT y
+PAX_MPROTECT_COMPAT n
+PAX_ELFRELOCS n
+PAX_KERNEXEC_PLUGIN_METHOD_BTS y # See PAX_RAP
+
+# PaX ASLR
+PAX_ASLR y
+PAX_RANDKSTACK y # Kernel Task Stack
+PAX_RANDUSTACK y # User Stack
+PAX_RANDMMAP y # Mmap without a base
+
+# PaX Miscellaneous
+PAX_MEMORY_SANITIZE y
+PAX_MEMORY_STACKLEAK y # bzero stack
+PAX_MEMORY_STRUCTLEAK y # bzero structs
+PAX_REFCOUNT y # (re)count references
+PAX_USERCOPY y # apply bounds copy_{from, to}_user
+PAX_SIZE_OVERFLOW y # assert size signature on function arguments
+PAX_SIZE_OVERFLOW_EXTRA n # do extra dynamic analysis
+PAX_INITIFY y # free unused kernel functions
+PAX_INITIFY_VERBOSE y # enable verbose mode to aid in debugging
+PAX_LATENT_ENTROPY y # generate more entropy at boot
```

Comment:
Enabling latent entropy instruments various functions to perform entropy gathering at runtime. The utility of doing that on a general-purpose kernel is debatable.